pkg/operator/podsecurityreadinesscontroller/violation.go

```diff
@@ -2,42 +2,36 @@ package podsecurityreadinesscontroller
 
 import (
 	"context"
+	"fmt"
 
+	securityv1 "github.com/openshift/api/security/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/sets"
 	applyconfiguration "k8s.io/client-go/applyconfigurations/core/v1"
-	"k8s.io/klog/v2"
 	psapi "k8s.io/pod-security-admission/api"
 )
 
 const (
 	syncerControllerName = "pod-security-admission-label-synchronization-controller"
 )
 
-var (
-	alertLabels = sets.New(psapi.WarnLevelLabel, psapi.AuditLevelLabel)
-)
-
 func (c *PodSecurityReadinessController) isNamespaceViolating(ctx context.Context, ns *corev1.Namespace) (bool, error) {
 	nsApplyConfig, err := applyconfiguration.ExtractNamespace(ns, syncerControllerName)
 	if err != nil {
 		return false, err
 	}
 
-	viableLabels := map[string]string{}
-	for alertLabel := range alertLabels {
-		if value, ok := nsApplyConfig.Labels[alertLabel]; ok {
-			viableLabels[alertLabel] = value
-		}
-	}
-	if len(viableLabels) == 0 {
-		// If there are no labels managed by the syncer, we can't make a decision.
-		return false, nil
+	var enforceLabel string
+
+	if _, ok := nsApplyConfig.Annotations[securityv1.MinimallySufficientPodSecurityStandard]; ok {
+		// Pick the MinimallySufficientPodSecurityStandard if it exists
+		enforceLabel = nsApplyConfig.Annotations[securityv1.MinimallySufficientPodSecurityStandard]
+	} else {
+		return false, fmt.Errorf("unable to determine if the namespace is violating because the MinimallySufficientPodSecurityStandard annotation wasn't found")
```

Comment:
Nit:
```go
	enforceLabel, ok := nsApplyConfig.Annotations[securityv1.MinimallySufficientPodSecurityStandard]
        if !ok {
		return false, fmt.Errorf("")
	}
```

Reply:
I am not explicitly sure if we would like to try to keep the previous logic as a backup. At least in the context of an upgrade.